synapse/rest/media/v1/preview_url_resource.py

```diff
@@ -504,20 +504,22 @@ async def _expire_url_cache_data(self) -> None:
             fname = self.filepaths.url_cache_filepath(media_id)
             try:
                 os.remove(fname)
+            except FileNotFoundError:
+                pass  # If the path doesn't exist, meh
             except OSError as e:
-                # If the path doesn't exist, meh
-                if e.errno != errno.ENOENT:
-                    logger.warning("Failed to remove media: %r: %s", media_id, e)
-                    continue
+                logger.warning("Failed to remove media: %r: %s", media_id, e)
+                continue
 
             removed_media.append(media_id)
 
-            try:
-                dirs = self.filepaths.url_cache_filepath_dirs_to_delete(media_id)
-                for dir in dirs:
+            dirs = self.filepaths.url_cache_filepath_dirs_to_delete(media_id)
+            for dir in dirs:
+                try:
                     os.rmdir(dir)
-            except Exception:
-                pass
+                except FileNotFoundError:
+                    pass  # Already deleted, continue with deleting the rest
+                except Exception:
+                    break  # Failed, skip deleting the rest of the parent dirs
```

Comment:
we should probably log something here?

Reply:
Sounds like a good idea. I got tired of the duplicated logic and factored it out into a function.